### PR TITLE
Dlist sync

### PR DESCRIPTION
**Pull request by lintal**
Lambda function to process an export of GAL & IBMS email subscription data and import this into the emergency-path.

Future changes:
* Alarms on the lambda.
* Audit logging of changes.

### CODE REVIEW

File: email-dlist-sync/index.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      currentAuthSenders,
+      currentSubscriptions,
+      ibmsgalExport,
+    ] = await Promise.all(fetchPromises);
+
+    const authSendersChangeset = new AuthorisedSenderChangeset(currentAuthSenders, ibmsgalExport);
+    const subsChangeset = new SubscriptionsChangeset(currentSubscriptions, ibmsgalExport);
+
+    if (!authSendersChangeset.isSafe()) {
+      errorHandler.handleHard(`The detected varance between the old and new authorised-senders lists is too great. Must be less than ${global.environment.maxVariance_AuthorisedSenders} but variance was ${authSendersChangeset.getPercentageVariance()}.`);
```
Comment (jcable) [inline email-dlist-sync/index.js]: varance -> variance

File: email-dlist-sync/lib/AuthorisedSenders/Adapter.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+class AuthorisedSendersAdapter {
+  constructor() {
+    this.s3 = new AWS.S3();
+
+    this.BucketName = global.environment.S3_AuthorisedSendersBucket;
+    this.ObjectKey = global.environment.S3_AuthorisedSendersObjectKey;
+  }
+
+  get() {
```
Comment (jcable) [inline email-dlist-sync/lib/AuthorisedSenders/Adapter.js]: Any advantage in this over:

return new AuthorisedSenders(JSON.parse((await this.s3.getObject( ... ).promise()).Body.toString()))

File: email-dlist-sync/index.js (hunk 1 above)
Comment (lintal) [reply email-dlist-sync/index.js]: Correction in #bef582e

File: email-dlist-sync/lib/AuthorisedSenders/Adapter.js (hunk 2 above)
Comment (lintal) [reply email-dlist-sync/lib/AuthorisedSenders/Adapter.js]: It just helps keep the main index.handler a bit tidier. Also you may note that I have a persist function which needs knowledge of the bucket in the putObject operation.

And I am also thinking ahead for the future, that I may want to substitute S3 for DynamoDB. Structuring in this way, I only need to change the adapter class and not really any other area of the lambda in theory.

**Pull request merged**